### PR TITLE
Enzyme tests for permissions provider

### PR DESCRIPTION
**Pull request by prabhjot-msft**
# What
Enzyme tests for permissions provider

# Why
https://skype.visualstudio.com/SPOOL/_workitems/edit/2916761
https://skype.visualstudio.com/SPOOL/_workitems/edit/2916762

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/permissions/PermissionsProvider.test.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  beforeEach(() => {
+    registerIcons({
+      icons: {
+        controlbuttoncameraoff: <></>,
+        chevrondown: <></>,
+        controlbuttoncameraon: <></>
+      }
+    });
+  });
+  test('Camera button is enabled for prsenter role', async () => {
```
Comment (alkwa-msft) [inline packages/react-components/src/permissions/PermissionsProvider.test.tsx:48]: presenter

**Review by alkwa-msft — APPROVED** [review]

File: packages/react-components/src/permissions/PermissionsProvider.test.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+describe('PermissionProvider tests for camera by passing mock permissions', () => {
+  beforeEach(() => {
+    registerIcons({
+      icons: {
+        controlbuttoncameraoff: <></>,
+        chevrondown: <></>,
+        controlbuttoncameraon: <></>
+      }
+    });
```
Comment (PorterNan) [inline packages/react-components/src/permissions/PermissionsProvider.test.tsx:46]: What do we do if we don't registerIcons? Will it throw erros?

File: packages/react-components/src/permissions/PermissionsProvider.test.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    expect(cameraButton.prop('disabled')).toBe(false);
+  });
+
+  test('Camera button is disabled for Consumer role', async () => {
+    const wrapper = mountWithPermissions(<CameraButton showLabel={true} />, mockConsumerPermissions);
+    const cameraButton = wrapper.find(ControlBarButton).first();
+    expect(cameraButton.prop('disabled')).toBe(true);
+  });
+});
+
```
Comment (PorterNan) [inline packages/react-components/src/permissions/PermissionsProvider.test.tsx:65]: Do we have a way to test hook as well? test the return value of usePermissions() separately
Like this one: https://stackoverflow.com/questions/65039773/how-to-test-custom-hooks-in-react-using-jest-enzyme, would be nice to only test hook and permissionProvider itself without existing component

File: packages/react-components/src/permissions/PermissionsProvider.test.tsx (hunk 2 above)
Comment (prabhjot-msft) [reply packages/react-components/src/permissions/PermissionsProvider.test.tsx:46]: Yes, it throws warnings and even adds delay to the tests
Screen Shot 2022-08-05 at 3 44 00 PM
.

File: packages/react-components/src/permissions/PermissionsProvider.test.tsx (hunk 1 above)
Comment (prabhjot-msft) [reply packages/react-components/src/permissions/PermissionsProvider.test.tsx]: done

File: packages/react-components/src/permissions/PermissionsProvider.test.tsx (hunk 3 above)
Comment (prabhjot-msft) [reply packages/react-components/src/permissions/PermissionsProvider.test.tsx:59]: Done

File: packages/react-components/src/permissions/PermissionsProvider.test.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+describe('Tests for permission provider hook by passing mock permissions', () => {
+  test('Test _usepermissions for presenter', async () => {
+    const wrapper = ({ children }): React.ReactElement => (
+      <PermissionsProvider permissions={mockPresenterPermissions}>{children}</PermissionsProvider>
+    );
+    const { result } = renderHook(() => _usePermissions(), { wrapper });
+    expect(result.current).toEqual(mockPresenterPermissions);
+  });
+  test('Test _usepermissions for attendee', async () => {
```
Comment (PorterNan) [inline packages/react-components/src/permissions/PermissionsProvider.test.tsx:44]: _usePermissions

File: packages/react-components/src/permissions/PermissionsProvider.test.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+const mockAttendeePermissions: _Permissions = {
+  cameraButton: true,
+  microphoneButton: true,
+  screenShare: false,
+  participantList: true,
+  removeParticipantButton: false
+};
+
+describe('Tests for permission provider hook by passing mock permissions', () => {
+  test('Test _usepermissions for presenter', async () => {
```
Comment (PorterNan) [inline packages/react-components/src/permissions/PermissionsProvider.test.tsx:37]: _usePermissions

File: packages/react-components/src/permissions/PermissionsProvider.test.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    expect(result.current).toEqual(mockPresenterPermissions);
+  });
+  test('Test _usepermissions for attendee', async () => {
+    const wrapper = ({ children }): React.ReactElement => (
+      <PermissionsProvider permissions={mockAttendeePermissions}>{children}</PermissionsProvider>
+    );
+    const { result } = renderHook(() => _usePermissions(), { wrapper });
+    expect(result.current).toEqual(mockAttendeePermissions);
+  });
+  test('Test _usepermissions for consumer', async () => {
```
Comment (PorterNan) [inline packages/react-components/src/permissions/PermissionsProvider.test.tsx:51]: _usePermissions

**Review by PorterNan — APPROVED** [review]
Nice, now we have an entirely independent test file only for provider and hooks

**Pull request merged**